java/component/keyple-plugin/remote-se/core/src/main/java/org/eclipse/keyple/plugin/remotese/core/KeypleMessageDto.java

```diff
@@ -0,0 +1,240 @@
+/********************************************************************************
+ * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
+ *
+ * See the NOTICE file(s) distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.keyple.plugin.remotese.core;
+
+/**
+ * This POJO object contains data exchanged between **Native SE** and **Virtual SE** components.
+ * <p>
+ * It is built and processed by the plugin and you don't need to modified it.
+ * <p>
+ * You only need to transfer it via the network by serializing and deserializing it on your own.<br>
+ * However, you can extend it or encapsulate it in a personal object if you need to transport other
+ * technical information related to the network infrastructure for example.
+ *
+ * @since 1.0
+ */
+public class KeypleMessageDto {
+
+    private String sessionId;
+    private String action;
+    private String clientNodeId;
+    private String serverNodeId;
+    private String nativeReaderName;
+    private String virtualReaderName;
+    private String body;
+    private String errorCode;
+    private String errorMessage;
+
+    /**
+     * Gets the session id.<br>
+     * In case of a full duplex communication, this field will permit to client and server to
+     * identify the socket.<br>
+     * This id is also useful for debugging.
+     *
+     * @return a not empty string.
+     * @since 1.0
+     */
+    public String getSessionId() {
+        return sessionId;
+    }
+
+    /**
+     * This setter method must only be used during the deserialization process.
+     *
+     * @param sessionId The session id to set.
+     * @return the object instance.
+     * @since 1.0
+     */
+    public KeypleMessageDto setSessionId(String sessionId) {
```

Comment:
I would rather have constructors than setters to make keypleMessageDto a immutable object